canals/pipeline/pipeline.py

```diff
@@ -337,6 +341,18 @@ def get_component(self, name: str) -> Component:
         except KeyError as exc:
             raise ValueError(f"Component named {name} not found in the pipeline.") from exc
 
+    def inputs(self):
+        """
+        Returns a dictionary with the input names and types that this pipeline accepts.
+        """
+        return _describe_pipeline_inputs(self.graph)
+
+    def print_inputs(self):
+        """
+        Prints a description of the input names and types that this pipeline accepts.
+        """
+        return _describe_pipeline_inputs_as_string(self.graph)
```

Comment:
I don't think we need this and it shouldn't be part of the scope of this PR, let's remove it.
We can add string representation later on and in a much nicer way too.

Reply:
I'll keep the function though, it's used in error messages right now. I'll make a utility in another PR. I want to use this representation in demos and presentations, the dictionary output is not as nice.

Reply:
Ah yes, leave it. It removes some duplication at least. I didn't comment about that on purpose.